symfony/assets/scripts/class/Visitor.ts

```diff
@@ -0,0 +1,37 @@
+import Storage from "./Storage";
+
+const STORAGE_KEY_LAST_VISITED_EVENTS_PAGE = "visitor/lastVisitedEventsPage";
+
+class Visitor {
+  public static get lastVisitedEventsPage(): Date | null {
+    const timestamp = Storage.getString(
+      STORAGE_KEY_LAST_VISITED_EVENTS_PAGE,
+      "",
+    );
+    if (!timestamp) {
+      return null;
+    }
+
+    // Ensure the string is actually a timestamp
+    const date = new Date(timestamp);
+    if (isNaN(date.valueOf())) {
+      return null;
+    }
+
+    return date;
+  }
+
+  public static set lastVisitedEventsPage(date: Date | null) {
+    if (!date) {
+      Storage.remove(STORAGE_KEY_LAST_VISITED_EVENTS_PAGE);
+      return;
+    }
+
+    Storage.saveString(
+      STORAGE_KEY_LAST_VISITED_EVENTS_PAGE,
+      date.toISOString(),
+    );
+  }
+}
+
+export default Visitor;
```

Comment:
Question from a non-experienced dev. This is a single-purpose file. Why didn't you merge this into `export default class Visitor {`? Is this a convention?

Reply:
Either approach is valid! It's just my personal preference to always do `export default` at the end of the file.

The reason I chose to do that is because when you compose the default export in another function, if you do that a couple of times, it starts affecting the indentation. You would see this a _lot_ in legacy class components React:
```typescript
// FILE: MyApp.tsx
export default connectRedux(myFunc)(
  withAuth(
    withCache(
      withFeatureFlags(
        class MyApp extends React.Component {
        }
      )
    )
  )
);
```
That meant that every time you added another function, it would completely reformat the whole file, leading to a lot of Git issues. By comparison, you don't have that same issue when it's `export default` at the end of the file.

I also find it just a little easier to read through the file and figure out what's the default export when it's always in a consistent location, vs. when it's mixed in with the file.

Ultimately though, this is just my personal preference. I don't think there's any real majority convention in JavaScript.